bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -376,6 +376,9 @@ def __init__(self, read_method, request, retry=DEFAULT_RETRY_READ_ROWS):
         self.rows = {}
         self._state = self.STATE_NEW_ROW
 
+        # Flag to stop iteration, for any reason not related to self.retry()
+        self.stop = False
```

Comment:
This needs to be set on cancel.  Also, this should be `self._stop`.

Reply:
Yes, good catch, thanks!
Done.